src/modules/EdgeDetect/EdgeUtils.js

```diff
@@ -1,174 +1,201 @@
-const _ = require('lodash')
-
-//define kernels for the sobel filter
-const kernelx = [[-1, 0, 1], [-2, 0, 2], [-1, 0, 1]],
-    kernely = [[-1, -2, -1], [0, 0, 0], [1, 2, 1]];
-
-module.exports = function(pixels, highThresholdRatio, lowThresholdRatio, inBrowser) {
-    let angles = [], mags = [], strongEdgePixels = [], weakEdgePixels = [], notInUI = !inBrowser;
-    for (var x = 0; x < pixels.shape[0]; x++) {
-        angles.push([]);
-        mags.push([]);
-        for (var y = 0; y < pixels.shape[1]; y++) {
-            var result = changePixel(
-                pixels,
-                pixels.get(x, y, 0),
-                pixels.get(x, y, 3),
-                x,
-                y
-            );
-            let pixel = result.pixel;
-
-            pixels.set(x, y, 0, pixel[0]);
-            pixels.set(x, y, 1, pixel[1]);
-            pixels.set(x, y, 2, pixel[2]);
-            pixels.set(x, y, 3, pixel[3]);
-
-            mags.slice(-1)[0].push(pixel[3]);
-            angles.slice(-1)[0].push(result.angle);
-        }
-    }
-    nonMaxSupress(pixels, mags, angles);
-    doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, mags, strongEdgePixels, weakEdgePixels);
-    return pixels;
-}
-
-//changepixel function that convolutes every pixel (sobel filter)
-function changePixel(pixels, val, a, x, y) {
-    let magX = 0.0;
-    for (let a = 0; a < 3; a++) {
-        for (let b = 0; b < 3; b++) {
-
-            let xn = x + a - 1;
-            let yn = y + b - 1;
-
-            magX += pixels.get(xn, yn, 0) * kernelx[a][b];
-        }
+// Define kernels for the sobel filter
+const kernelx = [
+  [-1, 0, 1],
+  [-2, 0, 2],
+  [-1, 0, 1]
+],
+kernely = [
+  [-1,-2,-1],
+  [ 0, 0, 0],
+  [ 1, 2, 1]
+];
+
+let pixelsToBeSupressed = [];
+
+module.exports = function(pixels, highThresholdRatio, lowThresholdRatio) {
+  let angles = [], grads = [], strongEdgePixels = [], weakEdgePixels = [];
+  for (var x = 0; x < pixels.shape[0]; x++) {
+    grads.push([]);
+    angles.push([]);
+    for (var y = 0; y < pixels.shape[1]; y++) {
+      var result = sobelFilter(
+        pixels,
+        x,
+        y
+      );
+      let pixel = result.pixel;
+
+      grads.slice(-1)[0].push(pixel[3]);
+      angles.slice(-1)[0].push(result.angle);
     }
-    let magY = 0.0;
-    for (let a = 0; a < 3; a++) {
-        for (let b = 0; b < 3; b++) {
+  }
+  nonMaxSupress(pixels, grads, angles);
+  doubleThreshold(pixels, highThresholdRatio, lowThresholdRatio, grads, strongEdgePixels, weakEdgePixels);
+  hysteresis(strongEdgePixels, weakEdgePixels);
```

Comment:
Here, we'd say `if (options.hysteresis)` ?

And in the info.json we'd set that as an option defaulting to true i guess?

Reply:
Yes. Exactly. It can default to false as well. Maybe we can default it to false since this module can crash old browsers if used by mistake. The browser will simply freeze. Maybe we can default it to true after alternate thread processing is implemented. Also the option it gets from the info.json can be a string and not a boolean and can also have a capital letter at the start so it might get itself messed up. Using
```js
options.hysteresis.toLowerCase() == 'true'
```
Is beeter I guess.